frontend/src/components/shared/Tooltip/Tooltip.tsx

```diff
@@ -54,6 +55,9 @@ function Tooltip({
 }: TooltipProps): ReactElement {
   const theme: Theme = useTheme()
   const { colors, fontSizes } = theme
+  const { activeTheme } = React.useContext(AppContext)
+  const tooltipBg =
+    activeTheme.name === "Light" ? colors.bgColor : colors.secondaryBg
```

Comment:
I don't feel confident relying on `activeTheme` name here. A preferred way would be to decide based on the luminance (we are doing this with a few other components): e.g. 
https://github.com/streamlit/streamlit/blob/366d63570c65e5465af59ef873a09625f7abe2ce/frontend/src/components/elements/Json/Json.tsx#L59

Since `tooltipBg` is only used once, this can be also just moved to directly down:

```typescript
backgroundColor: getLuminance(colors.bgColor) > 0.5 ? colors.bgColor : colors.secondaryBg
```

Reply:
+1 to this suggestion.

I also think it'd be worth considering adding a small helper function (that really just calls `getLuminance` on the theme background color) called `hasDarkBackgroundColor`, `hasLightBackgroundColor`, or something like that to make it more clear what's going on here. The `getLuminance` method is starting to appear in a bunch of places in the codebase so could be good to generalize in a more readable way

Reply:
Hi both! Thanks for the input. @LukasMasuch ' suggestion had been addressed in [8ee646d](https://github.com/streamlit/streamlit/pull/5275/commits/8ee646d78ebcd3c29f118a9ea7a7fecf20cab12d).

@vdonato I've also addressed your suggestion on [1fb07ef](https://github.com/streamlit/streamlit/pull/5275/commits/1fb07efdccf96260db75953c7bcab2629a55b6f4). Let me know if that's the correct place to add it, or if it should live somewhere else. Also, happy to update all our `getLuminance` calls with this new function if all looks good, let me know!

Reply:
Whoops, sorry @imjuangarcia! I must have missed this @mention so never got back to you on it. Then, I noticed the comment while going through unclosed PRs 😅 

The issue is due to a circular dependency import that occurs in 1fb07efdccf96260db75953c7bcab2629a55b6f4 because `hasLightBackgroundColor` is imported directly from `"src/theme/utils"` rather than from `"src/theme"`. Everything in `"src/theme/utils"` eventually gets re-exported by `"src/theme"` anyway, so changing where the import is coming from should fix this issue

Reply:
@vdonato no worries, and thanks for looking into it! I've addressed this on [ead3064](https://github.com/streamlit/streamlit/pull/5275/commits/ead3064cffab037eb8d7d6c45fe782bd35809034), let me know if you need anything else from me here!